core/primitives/piece/impl/piece_data.cpp

```diff
@@ -26,16 +28,22 @@ namespace fc::primitives::piece {
     return fd_ != kUnopenedFileDescriptor;
   }
 
+  bool PieceData::getIsNullData() const {
+    return isNullData;
+  }
+
   PieceData::PieceData(PieceData &&other) noexcept
-      : fd_(kUnopenedFileDescriptor) {
+      : fd_(kUnopenedFileDescriptor), isNullData(other.isNullData) {
     fd_ = other.fd_;
     other.fd_ = kUnopenedFileDescriptor;
   }
 
-  PieceData::PieceData(int &pipe_fd) : fd_(pipe_fd) {
+  PieceData::PieceData(int &pipe_fd) : fd_(pipe_fd), isNullData(false) {
     pipe_fd = kUnopenedFileDescriptor;
   }
 
+  PieceData::PieceData() : isNullData(true) {}
```

Comment:
Move it upper. In general, all constructors are placed above other methods

Reply:
Fixed